packages/assets-controllers/src/AccountTrackerController.ts

```diff
@@ -282,69 +282,113 @@ export class AccountTrackerController extends StaticIntervalPollingController<Ac
-        ? Object.keys(accountsByChainId[chainId])
-        : [toChecksumHexAddress(selectedAccount.address)];
-
-      const accountsForChain = { ...accountsByChainId[chainId] };
-      for (const address of accountsToUpdate) {
-        const balance = await this.#getBalanceFromChain(address, ethQuery);
-        if (balance) {
-          accountsForChain[address] = {
-            balance,
-          };
-        }
-        if (this.#includeStakedAssets) {
-          const stakedBalance = await this.#getStakedBalanceForChain(
-            address,
-            networkClientId,
+      // Create an array of promises for each networkClientId
+      const updatePromises = networkClientIds.map(async (networkClientId) => {
+        const { chainId, ethQuery } =
+          this.#getCorrectNetworkClient(networkClientId);
+        this.syncAccounts(chainId);
+        const { accountsByChainId } = this.state;
+        const { isMultiAccountBalancesEnabled } = this.messagingSystem.call(
+          'PreferencesController:getState',
+        );
+
+        const accountsToUpdate = isMultiAccountBalancesEnabled
+          ? Object.keys(accountsByChainId[chainId])
+          : [toChecksumHexAddress(selectedAccount.address)];
+
+        const accountsForChain = { ...accountsByChainId[chainId] };
+
+        // Create an array of promises for balance and staked balance fetching
+        const balancePromises = accountsToUpdate.map(async (address) => {
+          const balancePromise = this.#getBalanceFromChain(address, ethQuery);
+          const stakedBalancePromise = this.#includeStakedAssets
+            ? this.#getStakedBalanceForChain(address, networkClientId)
+            : Promise.resolve(null);
+
+          const [balanceResult, stakedBalanceResult] = await Promise.allSettled(
+            [balancePromise, stakedBalancePromise],
           );
-          if (stakedBalance) {
+
+          // Update account balances
+          if (balanceResult.status === 'fulfilled' && balanceResult.value) {
+            accountsForChain[address] = {
+              balance: balanceResult.value,
+            };
+          }
+
+          if (
+            stakedBalanceResult.status === 'fulfilled' &&
+            stakedBalanceResult.value
+          ) {
             accountsForChain[address] = {
               ...accountsForChain[address],
-              stakedBalance,
+              stakedBalance: stakedBalanceResult.value,
             };
           }
-        }
-      }
+        });
 
-      this.update((state) => {
-        state.accountsByChainId[chainId] = accountsForChain;
+        // Wait for all balance-related promises to settle
+        await Promise.allSettled(balancePromises);
+
+        // After all promises for this networkClientId are settled, return the updated data
+        return { chainId, accountsForChain };
+      });
+
+      // Wait for all networkClientId updates to settle in parallel
+      const allResults = await Promise.allSettled(updatePromises);
```

Comment:
Hmmm so we will make parallel async calls for: each network * each address * 2 (balance result and staked balance result).
And this polling is every 10 seconds.

I'm wondering if this is quite expensive.

Reply:
Discussed this, it is the original behaviour from before.
We can do further discussions as a team on how we can further improve this.